src/test/java/io/github/oliviercailloux/y2018/apartments/apartment/json/JsonConvertTest.java

```diff
@@ -84,12 +98,37 @@ void jsonToApartmentsTest() throws IOException, URISyntaxException {
             .setTele(false)
             .build());
 
-    URI ressource = JsonConvertTest.class.getResource("jsonApartments.json").toURI();
-    Path jsonPath = Path.of(ressource);
-
-    List<Apartment> apartmentsTest = JsonConvert.jsonToApartments(jsonPath);
+    List<Apartment> apartmentsTest = JsonConvert.jsonToApartments(this.jsonTestPath);
 
     assertEquals(apartmentsRef.get(0).hashCode(), apartmentsTest.get(0).hashCode());
     assertEquals(apartmentsRef.get(1).hashCode(), apartmentsTest.get(1).hashCode());
   }
+
+  /** Test if the adapter method works */
+  @Test
+  void getAdapterTest() throws Exception {
+    JsonbConfig defaultConfig = new JsonbConfig();
+    JsonbAdapter<Apartment, JsonObject> adapter = JsonConvert.getAdapter();
+    JsonbConfig config = new JsonbConfig().withAdapters(adapter);
+    assertNotEquals(defaultConfig, config);
+    assertThrows(
+        UnsupportedOperationException.class,
+        () -> adapter.adaptToJson(null),
+        "The function should not be implemented");
```

Comment:
Ceci ne fait que répéter ce que vous avez déjà dit : votre code de test n’est pas vraiment différent du code original. Il ne peut pas échouer, si on ne change pas fondamentalement le code original. OPT

Reply:
Nous avons en effet eu du mal à déterminer les TU nécessaires pour l'`adapter` (nous avons hésité à en faire).
En ce sens, la fonction est spécifique et l'implémentation est relativement banale/simple.

Reply:
Fix in 2fd4cc3729016854db9dc872f869b9b221bd4ac2